dotnet/Capstone/Controllers/BirdController.cs

```diff
@@ -71,35 +73,117 @@ public Bird getRandomBird()
+            const string errorMessage = "An error occurred and a bird sighting was not created.";
+            
+            IActionResult result;
             try
             {
-                Bird bird = birdDao.createBird(newBird, newBird.name, newBird.description, newBird.imgUrl);
-                if (bird == null)
-                {
-                    Console.WriteLine(StatusCode(404));
-                }
-            }catch (DaoException )
+                Bird bird = birdDao.createBird(newBird, listId);
+
+                result = Created("", "");
+            }
+            catch (DaoException e)
             {
-                Console.WriteLine(StatusCode(500));
+                result = BadRequest(new { message = e.Message });
             }
+            catch (ArgumentException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch
+            {
+                result = BadRequest(new { message = errorMessage });
+            }
+
+            return result;
 
         }
 
-        [HttpPut("/birds/{id}")]
-        public void editBird(Bird updatedBird, int id)
+        [HttpPut("/updateBird/{id}")]
+        public IActionResult editBird(Bird updatedBird, int id)
         {
-            birdDao.editBird(updatedBird, id);
+            const string errorMessage = "An error occurred and bird could not be modified.";
+            IActionResult result;
+            try
+            {
+                birdDao.editBird(updatedBird, id);
+                result = Ok();
+            }
+            catch (DaoException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch (ArgumentException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch
+            {
+                result = BadRequest(new { message = errorMessage });
+            }
+
+
+            return result;
 
         }
 
-        [HttpDelete("/birds/{id}")]
-        public void deleteBird(int id)
+        [HttpDelete("/deleteBird/{id}")]
+        public IActionResult deleteBird(int id)
         {
-            birdDao.deleteBird(id);
+            
+            const string errorMessage = "An error occurred and bird could not be deleted.";
+            IActionResult result;
+            try
+            {
+                birdDao.deleteBird(id);
+                result = NoContent();
+            }
+            catch (DaoException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch (ArgumentException e)
+            {
+                result = BadRequest(new { message = e.Message });
+            }
+            catch
+            {
+                result = BadRequest(new { message = errorMessage });
+            }
+
+
+            return result;
+        }
+
+ 
+
+
+        [HttpGet("/lists/{listId}/birds")]
```

Comment:
Is there a reason this is in the bird controller and not the list controller? I think I follow, but I just wanted to be sure it had been considered.

Reply:
Yes, this is because it is getting birds that share the same list id. The reason why the URL is that way is to match the front-end BirdService in vue.

Reply:
I will update the README either today or tomorrow if I have some time, but other than that, everything is done. Feel free to check and let me know if there is something I should check again.